sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java

```diff
@@ -0,0 +1,949 @@
+                                .then(Mono.just(s));
+                        }
+                    }
+                }
+                return ret;
+            });
+        });
+    }
+
+    /**
+     * Gets the connection ID.
+     *
+     * @return the connection ID.
+     */
+    public String getConnectionId() {
+        return connectionId;
+    }
+
+    /**
+     * Starts the client for connecting to the server.
+     *
+     * @return the task.
+     */
+    public Mono<Void> start() {
+        if (clientState.get() != WebPubSubClientState.STOPPED) {
+            return Mono.error(logger.logExceptionAsError(
+                new IllegalStateException("Failed to start. Client is not STOPPED.")));
+        }
+        return Mono.defer(() -> {
+            // reset
+            isStoppedByUser.set(false);
+            sequenceAckId.clear();
+
+            boolean success = clientState.changeStateOn(WebPubSubClientState.STOPPED, WebPubSubClientState.CONNECTING);
+            if (!success) {
+                return Mono.error(logger.logExceptionAsError(
+                    new IllegalStateException("Failed to start. Client is not STOPPED.")));
+            } else {
+                return Mono.empty();
+            }
+        }).then(clientAccessUriProvider.flatMap(uri -> Mono.fromCallable(() -> {
+            this.endpoint = new ClientEndpoint();
+            ClientEndpointConfig config = ClientEndpointConfig.Builder.create()
+                .preferredSubprotocols(Collections.singletonList(webPubSubProtocol.getName()))
+                .encoders(Collections.singletonList(MessageEncoder.class))
+                .decoders(Collections.singletonList(MessageDecoder.class))
+                .build();
+            this.session = clientManager.connectToServer(endpoint, config, new URI(uri));
+            return (Void) null;
+        }).subscribeOn(Schedulers.boundedElastic()))).doOnError(error -> {
+            handleClientStop();
+        });
+    }
+
+    /**
+     * Stops the client for disconnecting from the server.
+     *
+     * @return the task.
+     */
+    public Mono<Void> stop() {
+        if (clientState.get() == WebPubSubClientState.CLOSED) {
+            return Mono.error(logger.logExceptionAsError(
+                new IllegalStateException("Failed to stop. Client is CLOSED.")));
+        }
+        return Mono.defer(() -> {
+            // reset
+            isStoppedByUser.set(true);
+            isStoppedByUserMono.set(null);
+            groups.clear();
+
+            if (session != null && session.isOpen()) {
+                return Mono.fromCallable(() -> {
+                    session.close(CloseReasons.NO_STATUS_CODE.getCloseReason());
+                    return (Void) null;
+                }).subscribeOn(Schedulers.boundedElastic());
+            } else {
+                if (clientState.clientState.get() == WebPubSubClientState.STOPPED) {
```

Comment:
nit: use clientState.get() instead of clientState.clientState.get() align with code above.

Reply:
thx, fixed.